sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -351,11 +399,54 @@ export function getRawSqlParameters(parameters: SqlParameter[] | undefined): any
 /**
  * @internal
  * @ignore
- * Helper utility to build an instance of raw SQL parameter as `RawSqlParameter`
+ * Helper utility to extract array of userProperties key:value instances from given input,
+ * or undefined if not passed in.
+ * @param value
+ */
+export function getRawUserProperties(parameters: { [key: string]: any } | undefined): any {
+  if (parameters == undefined) {
+    return undefined;
+  }
+
+  const rawParameters: RawKeyValuePair[] = [];
+  for (let [key, value] of Object.entries(parameters)) {
+    let type: string | number;
+    if (typeof value === "number") {
+      type = TypeMapForRequestSerialization.number;
+    } else if (typeof value === "string") {
+      type = TypeMapForRequestSerialization.string;
+    } else {
+      throw new TypeError(
+        `Unsupported type for the value in the user property {${key}:${JSON.stringify(value)}}`
+      );
+      // TODO: Add more cases if other types are supported.
+    }
+    const rawParameterValue: any = {};
```

Comment:
I know sometimes in deserialization/serialization code it's tempting to go to 'any' but it seems like you have a well defined field name that you could just make part of an interface type.

```typescript
{
  // or just the field name. Not sure what that constant is.
  [Constants.XML_METADATA_MARKER]: any;    // although it'd be awesome if we could add types for these fields as well.
}
```

Reply:
Didn't put much thought into it while getting it to work, I'll update this to something beautiful.

Reply:
Updated to not have any "any"s.